File: apps/ios/Pebbles/Features/Path/Models/WeekRollEntry.swift
import Foundation

/// One slot in the iOS Path's horizontal weeks roll.
///
/// `weekStart` is the ISO Monday 00:00 of the week, in the calendar passed
/// to `WeekRollBuilder.build(...)`. `pebbles` is already sorted per the
/// past-vs-current rule (past = oldest first, current/future = newest first).
///
/// Identity is `weekStart` so SwiftUI's `ForEach` keys correctly across
/// rebuilds (e.g. after a pebble create that does not change which weeks
/// are populated).
struct WeekRollEntry: Identifiable, Hashable {
    let weekStart: Date
    let pebbles: [Pebble]
    var id: Date { weekStart }
}

File: apps/android/app/src/main/kotlin/app/pbbls/android/features/path/models/WeekRollEntry.kt
package app.pbbls.android.features.path.models

import java.time.LocalDate

/**
 * One ISO week of the path — mirrors iOS `WeekRollEntry.swift`.
 *
 * @param weekStart the ISO Monday of the week, in the device zone.
 * @param pebbles pre-sorted by `WeekRollBuilder` (past weeks ascending,
 *   current/future descending by `happenedAt`).
 */
data class WeekRollEntry(
    val weekStart: LocalDate,
    val pebbles: List<Pebble>,
)
